src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityFurnace.java

```diff
@@ -69,14 +73,59 @@
     private boolean canIdle = false;
     private boolean isSmeltPaused = false;
     private long powerUsage = 50;
+    private MachineTier machineTier;
+
 
     @Override
     public void markForUpdate() {
         super.markForUpdate();
-
         this.markForLightUpdate();
     }
 
+    @Override
+    protected void syncDataTo(NBTTagCompound nbtTagCompound, SyncReason syncReason) {
+        super.syncDataTo(nbtTagCompound, syncReason);
+        nbtTagCompound.setFloat("internalTemp",internalTemp);
+        nbtTagCompound.setInteger("smeltProgress",smeltProgress);
+    }
+
+    @Override
+    protected void syncDataFrom(NBTTagCompound nbtTagCompound, SyncReason syncReason) {
+        super.syncDataFrom(nbtTagCompound, syncReason);
+        internalTemp = nbtTagCompound.getFloat("internalTemp");
+        smeltProgress = nbtTagCompound.getInteger("smeltProgress");
+    }
+
+    @Override
+    public boolean canExtractItem(int index, ItemStack stack, EnumFacing direction) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "canExtractItem" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)